### ISSUE TITLE
[BUG]: no output scaling on test samples in `chemprop train`

### ISSUE THREAD

**Issue by davidegraff** [chemprop/chemprop#512]
**Describe the bug**
When we train a model using `chemprop train`, we first normalize our train dataset and scale our validation dataset accordingly. We later use the pytorch lightning `Trainer.test` functionality on our _unscaled_ test set. **However**, we never unscale our predictions, so our test errors are wildly inflated.

**Example(s)**
In this block of `train.py`:

```python
if isinstance(readout_ffn, RegressionFFN):
    scaler = train_dset.normalize_targets()     
    val_dset.normalize_targets(scaler)
    logger.info(f"Train data: loc = {scaler.mean_}, scale = {scaler.scale_}")
```

we should be setting the `scale` and `loc` attributes of our `RegressionFFN`. However, doing so would also mess up validation. This is because validation examples flow through `forward()`, which performs an output transformation. We will need to either

1. not scale our validation set. Easy fix, but now our validation loss is not comparable to our train loss
2. not use `forward()` in the `validation_step`. There's no specific reason why this would be a problem, but the high-level idea when writing this originally was to make validation look more like testing than training.

**Expected behavior**
That the testing output is scaled appropriately by the fitted statistics of the training dataset.

**Comment by hwpang** [comment]
Is this addressed now with the following lines in `traing.py` around https://github.com/chemprop/chemprop/blob/v2/dev/chemprop/cli/train.py#L682?

```
    if test_loader is not None:
        if args.task_type == "regression":
            model.predictor.register_buffer("loc", torch.tensor(scaler.mean_).view(-1, 1))
            model.predictor.register_buffer("scale", torch.tensor(scaler.scale_).view(-1, 1))
        results = trainer.test(model, test_loader)[0]
        logger.info(f"Test results: {results}")
```

**Comment by davidegraff** [comment]
IMO (which I assume is one you share):
1. we should remove the `loc` and `scale` buffers from the `RegressionFFN`, then
2. one of:
    - tell users that they need to _unscale_ the predictions using the `train_dset`'s fitted `StandardScaler`
    - add an `MPNN.scaler: StandardScaler | None` attribute that scales the outputs of `predict_step()`, if the attribute exists.

I feel like just telling people to unscale is simpler and cleaner
